### ISSUE TITLE
visual slam tracks aligned with GPS

### ISSUE THREAD

**Issue by QMonteCarlo** [waiwnf/pilotguru#36]
I have use this app to take records, and generate tracks from the video, then align it with the GPS track. The typical error is about 10m. The altitude from GPS is most unreliable: even when I walk on a plane, the altitude fluctuates some 20 meters. 
I'll post the aligned tracks later, if someone wants.

**Comment by QMonteCarlo** [comment]
Recently I have changed my Android phone to Xiaomi 8, the first dual frequency GNSS smartphone.
The gps data quality is much better now, compared with previous old ones, however, position fluctuate still exist - after all, this is a consumer grade device.
Here is the visual trajectory aligned with gps, I have chosen to walk along known ground straight lines as a crude groundtruth.
For small area, visual slam with loop closing is more accurate.
![gps-visual-trajectory-align](https://user-images.githubusercontent.com/21305149/45129766-543a3600-b1b8-11e8-858d-7aae5f60e0bd.png)

altitude fluctuatation:
![gps-altitude-profile](https://user-images.githubusercontent.com/21305149/45129799-759b2200-b1b8-11e8-96c6-cbf4cd7db712.png)

**Comment by waiwnf** [comment]
Interesting, thanks for posting these! I am actually very impressed how well the optical SLAM has performed here. For my purposes (outdoors, challenging lighting, potentially quick motions) I never was able to get it to work robustly, so ended up giving up on SLAM in favor of other approaches, but great to see it working well for other use cases!

**Comment by QMonteCarlo** [comment]
you're right, visual slam is not so reliable , and that's what I am working on.
